cmd/kube-proxy/app/server.go

```diff
@@ -116,20 +157,69 @@ Service cluster ips and ports are currently found through Docker-links-compatibl
 environment variables specifying ports opened by the service proxy. There is an optional
 addon that provides cluster DNS for these cluster IPs. The user must create a service
 with the apiserver API to configure the proxy.`,
-		Run: func(cmd *cobra.Command, args []string) {
+		Run: func(c *cobra.Command, args []string) {
+			cmdutil.CheckErr(opts.Complete())
+			cmdutil.CheckErr(opts.Validate(args))
+			cmdutil.CheckErr(opts.Run())
 		},
 	}
 
+	flags := cmd.Flags()
+	BindKubeProxyOptions(&opts, flags)
+
+	cmd.MarkFlagFilename("config", "yaml", "yml")
+
 	return cmd
 }
 
-// NewProxyServerDefault creates a new ProxyServer object with default parameters.
-func NewProxyServerDefault(config *options.ProxyServerConfig) (*ProxyServer, error) {
+// ProxyServer represents all the parameters required to start the Kuberetes
+// proxy server. All fields are required.
+type ProxyServer struct {
+	Client                 *kubeclient.Client
+	IptInterface           utiliptables.Interface
+	Proxier                proxy.ProxyProvider
+	EndpointsHandler       proxyconfig.EndpointsConfigHandler
+	Broadcaster            record.EventBroadcaster
+	Recorder               record.EventRecorder
+	ConntrackConfiguration componentconfig.KubeProxyConntrackConfiguration
+	Conntracker            Conntracker // if nil, ignored
+	ProxyMode              string
```

Comment:
Is it intentional to duplicate all the config values here? Feels like a lot of bloat instead of just leaving the config reference in here (we have that at many places in other component as well, unfortunately).